### ISSUE TITLE
RaspberryPiOS Screen + Touch

### ISSUE THREAD

**Issue by hkfuertes** [tomasz-grobelny/AACS#36]
Quick question, How can I build/integrate this project into raspberrypi os? My idea is to have my phone on trips... but also have a "desktop" (rpi4) in my car that I could plug in for Jellyfin to stream content when parked...

**Comment by hkfuertes** [comment]
In fact... even just Kodi/OpenElec would suffice...

**Comment by bert1337** [comment]
I can confirm that this repo works in raspberry pi os.
I am running it in a pi zero 2w (just for 480p video, no de or x11 server running) and it works allright.
You would need to extract current keys form android auto app and apply the build fixes, but the rest should be good!
At the moment i'm not getting the inputs from the headunit so i'm not sure if that works but i can confirm that using vnc works well.

**Comment by hkfuertes** [comment]
so you have pios running on your car? do you have any guide?

**Comment by bert1337** [comment]
Yes, if you need to see the whole desktop you could use something like ximagesrc in your gst-launch-1.0 pipeline (without specificing the xname) the same that is used in the startx_internal script in this repo.

To get stuff working i just built the repo using the build fixes pr, modified a little the startx_internal script (as i did not need anbox).
If you just need to view the pi screen on the car you dont need AAClient, just the AAServer.
So your script must, in this order:
1. Setup the env variables (same as the startx_internal)
2. start snomix using the correct base.ini
3. start AAServer (maybe in a loop so it will restart on crash)
4. setup and launch the GStreamer pipeline

And obviously you would need current certs, you can extract them yourself or i can send you some that will work until mid october (your car could also just not check for them so i would try with older certs and check if ssl auth works)

**Comment by wes1993** [comment]
Hello @bert1337,
Thanks a lot for the detailed explanation!
Could you please send me the certificates you mentioned? Also, do you recommend using a specific version of Raspberry Pi OS for this setup?

Best regards
Stefano

**Comment by bert1337** [comment]
@wes1993 add me on discord (my username is beeeert) and i will share more info about the certs.
I dont remember the version of pi os i used but i dont think it metters too much, as i was able to make it work nicely on a debian 11 in the Radxa cubie A5E aswell on ubuntu 25, so i would not worrie about the os too much

**Comment by wes1993** [comment]
Hi @bert1337 I’ve just sent you a friend request on Discord. 
Thanks a lot for offering to share more info about the certs, I really appreciate it! 
Looking forward to learning from your experience.

Bye
Stefano

**Comment by gamelaster** [comment]
@wes1993 in another Issue of this AACS repo are instructions how to dump the certs.
